### PR TITLE
[CS] Align php-cs-fixer config and code with fixer update to v2.7.1

### PR DESCRIPTION
**Pull request by alongosz**
We recently updated php-cs-fixer to `v2.7.1`. This PR aligns `ezpublish-kernel` with that update.

**Backporting this to 6.7** as it is LTS branch, so we will likely need new rules to apply for PRs targeting that as well.

IMHO some of the rules make sense. I placed each fix in separate commit, easy to revert, if you, reviewers, don't agree with some of these rules.
**Please see [my review (comments)](#pullrequestreview-68854492) before reviewing.**

Rules doc: https://github.com/FriendsOfPHP/PHP-CS-Fixer/tree/v2.7.1#usage

Changes introduced by PR:
- [x] Removed php-cs-fixer config (in `.php_cs`) for v1.x as it's no longer needed
- [x] Added to `.php_cs` rule `yoda_style => false`
- [x] Added to `.php_cs` rule `no_break_comment => false`
- [x] Fixed code according to rule `modernize_types_casting`
- [x] Fixed code according to rule `function_to_constant`
- [x] Fixed code according to rule `no_unused_imports`
- [x] Fixed code according to rule `non_printable_character`
- [x] Fixed code according to rule `dir_constant`
- [x] Fixed code according to rule `braces` (surprisingly it fixes for us wrong indentation only)

### CODE REVIEW

File: eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php [hunk 1]
```diff
@@ -68,6 +68,6 @@ protected function execute(InputInterface $input, OutputInterface $output)
             throw new RuntimeException("'--iteration-count' option should be > 0, got '{$iterationCount}'");
         }
 
-        $this->searchIndexer->createSearchIndex($output, intval($iterationCount), empty($noCommit));
+        $this->searchIndexer->createSearchIndex($output, (int) $iterationCount, empty($noCommit));
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php]: This is `modernize_types_casting`. It applies only to code introduced by me, so it seems I was the only fan of `intval`. With the direction PHP7 is going to, now it makes more sense to apply normal casting.

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Configuration/Suggestion/Formatter/YamlSuggestionFormatter.php [hunk 2]
```diff
@@ -19,7 +19,7 @@ public function format(ConfigSuggestion $configSuggestion)
         $suggestion = $configSuggestion->getSuggestion();
         if ($suggestion) {
             $yamlConfig = Yaml::dump($suggestion, 8);
-            if (php_sapi_name() !== 'cli') {
+            if (PHP_SAPI !== 'cli') {
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Configuration/Suggestion/Formatter/YamlSuggestionFormatter.php]: Rule `function_to_constant`. I think also valuable for performance (microseconds, but still...!). Also it's the only one occurrence AFAIR

File: eZ/Publish/API/Repository/Tests/Regression/EZP20018VisibilityTest.php [hunk 3]
```diff
@@ -33,7 +33,7 @@ public function testSearchForHiddenContent()
         $this->assertEquals(0, $results1->totalCount);
         $this->assertCount(0, $results1->searchHits);
 
-         // Hide "Images" Folder
+        // Hide "Images" Folder
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/Regression/EZP20018VisibilityTest.php]: Don't know why this and similar indentation issues were matched by `braces` rule, but it's worth to align code properly.

File: eZ/Publish/Core/Limitation/BlockingLimitationType.php [hunk 4]
```diff
@@ -109,7 +109,7 @@ public function buildValue(array $limitationValues)
      * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException If any of the arguments are invalid
      *         Example: If LimitationValue is instance of ContentTypeLimitationValue, and Type is SectionLimitationType.
      * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException If value of the LimitationValue is unsupported
-     *         Example if OwnerLimitationValue->limitationValues[0] is not one of: [ 1,  2 ]
+     *         Example if OwnerLimitationValue->limitationValues[0] is not one of: [ 1,  2 ]
```
Comment (alongosz) [inline eZ/Publish/Core/Limitation/BlockingLimitationType.php]: Matched by `non_printable_character`. All of the changes are related to some whitespace characters outside of standard ASCII table.

File: eZ/Publish/Core/MVC/Symfony/Templating/Tests/Twig/Extension/ContentExtensionTest.php [hunk 5]
```diff
@@ -53,7 +53,7 @@ public function getExtensions()
 
     public function getFixturesDir()
     {
-        return dirname(__FILE__) . '/_fixtures/content_functions/';
+        return __DIR__ . '/_fixtures/content_functions/';
```
Comment (alongosz) [inline eZ/Publish/Core/MVC/Symfony/Templating/Tests/Twig/Extension/ContentExtensionTest.php]: Rule `dir_constant`. It also maybe brings some performance and readability.

File: eZ/Publish/Core/Repository/Tests/Service/Integration/RoleBase.php [hunk 6]
```diff
@@ -345,21 +345,21 @@ public function testUpdateRole()
...
+                    $updatedRole->getNames()
+                );
+
+                self::assertEquals(
+                    array(
+                        'eng-US' => $roleUpdateStruct->descriptions['eng-US']
+                    ),
+                    $updatedRole->getDescriptions()
+                );
+        */
```
Comment (alongosz) [inline eZ/Publish/Core/Repository/Tests/Service/Integration/RoleBase.php]: Code indentation as well
Note to self: commented-out, looks like some follow-up for the future

File: .php_cs [hunk 7]
```diff
@@ -74,6 +17,9 @@ return PhpCsFixer\Config::create()
         'phpdoc_annotation_without_dot' => false,
         'phpdoc_no_alias_tag' => false,
         'space_after_semicolon' => false,
+        'yoda_style' => false,
+        'modernize_types_casting' => false,
```
Comment (andrerom) [inline .php_cs]: you disabled this but some changes here is related to this rule right? `(int)` So should it be enabled or does it then change a lot more code?

Comment (alongosz) [reply .php_cs]: Ah, sorry, my mistake, I actually fixed code according to this rule and commented [here](#discussion_r144227994).
Removed it from .php_cs, updated PR.

Comment (andrerom) [reply .php_cs]: +1

**Pull request merged**